Only functionalize ops which we want to include in mkldnn group

eellison opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #52614 Avoid dispatch overhead in call to mkldnn convolution
* #52613 Include max pool in fusion groups
* #52513 Extend subgraph utils to cover merging a node following a subgraph
* #52512 Remove unused subgraph vmap api
* **#51924 Only functionalize ops which we want to include in mkldnn group**
* #51923 Add a filter to remove mutation
* #51922 Dont error on 0-dim in convolution
* #51921 Add 0-dim support for binary mkldnn ops
* #51736 Handle mkldnn broadcasting in mkldnn fuser
* #51600 Add MKLDNN fuser
* #52786 [NNC] Make NNC sanitize input names
* #51484 Add support for linear in mkldnn fusion
* #51483 Move conv to mkldnn

Differential Revision: [D26696705](https://our.internmc.facebook.com/intern/diff/D26696705)

Commit:
Only functionalize ops which we want to include in mkldnn group

[ghstack-poisoned]

Commit:
Update on "Only functionalize ops which we want to include in mkldnn group"




[ghstack-poisoned]

ZolotukhinM commented on test/jit/test_freezing.py:
```diff
@@ -1402,7 +1402,11 @@ def forward(self, x):
+            self.run_pass("inline", scripted_mod.graph)
+            print(scripted_mod.graph)
+            out = torch.jit._script.RecursiveScriptModule(torch._C._freeze_module(scripted_mod._c, preserveParameters=True))
+            print(out.graph)
+            return
```
Is it a leftover?

ZolotukhinM commented on test/jit/test_freezing.py:
```diff
@@ -1375,7 +1375,7 @@ def tearDown(self):
         conv_bias = [True, False]
         module_pairs = [(nn.Conv1d, nn.BatchNorm1d), (nn.Conv2d, nn.BatchNorm2d), (nn.Conv3d, nn.BatchNorm3d)]
-        use_tracing = [True, False]
+        use_tracing = [False]
```
Unintentional?

ZolotukhinM reviewed (approved):
Looks good, but please make sure the tests are in a good shape :)

Krovatkin reviewed (approved):
:shipit:

Commit:
Update on "Only functionalize ops which we want to include in mkldnn group"




[ghstack-poisoned]

Commit:
Update on "Only functionalize ops which we want to include in mkldnn group"




[ghstack-poisoned]

Commit:
Update on "Only functionalize ops which we want to include in mkldnn group"




[ghstack-poisoned]

Commit:
Update on "Only functionalize ops which we want to include in mkldnn group"




[ghstack-poisoned]

Commit:
Update on "Only functionalize ops which we want to include in mkldnn group"




[ghstack-poisoned]

Commit:
Update on "Only functionalize ops which we want to include in mkldnn group"




[ghstack-poisoned]

Commit:
Update on "Only functionalize ops which we want to include in mkldnn group"




[ghstack-poisoned]

Commit:
Update on "Only functionalize ops which we want to include in mkldnn group"




[ghstack-poisoned]

Commit:
Update on "Only functionalize ops which we want to include in mkldnn group"




[ghstack-poisoned]

Commit:
Update on "Only functionalize ops which we want to include in mkldnn group"


Differential Revision: [D26696705](https://our.internmc.facebook.com/intern/diff/D26696705)

[ghstack-poisoned]
